cfn-guard-lambda CentOS / RedHat builds

seawildau opened the pull request:
*Issue #, if available:*

*Description of changes:*

Added instructions and commands for CentOS / RedHat builds

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

nathanataws commented on cfn-guard-lambda/README.md:
```diff
@@ -18,6 +18,7 @@ The primary interface for building and deploying the tool is the [Makefile](Make
 * [Rust](https://rustup.rs/) (See the installation instructions in the [top-level README](../README.md#install-rust))
 * If building on a Mac, you'll need [Homebrew](https://brew.sh/).  
 * If building on Ubuntu, you'll need to run `sudo apt-get update; sudo apt install build-essential` if you haven't already
+* If building on CentOS/RHEL you'll need to install `jq`, `gcc` and the `musl-devel` package (see https://copr.fedorainfracloud.org/coprs/ngompa/musl-libc/ for instructions to add the musl-libc repository to your yum config)
```
Since the `make pre-reqs` command does the install here, could we just make this:

"you'll need to add the musl-libc repository to your yum config: <url>"

?

nathanataws commented on cfn-guard-lambda/Makefile:
```diff
@@ -135,11 +135,12 @@ pre-reqs:
+			sudo yum install jq gcc zip musl-devel
+      else ifeq "$(shell cat /etc/system-release | grep -v 'CentOS Linux')" ""
+			@echo Distro is CentOS.
+			sudo yum install jq gcc zip musl-devel
```
This is fantastic!  Thank you for doing this.  I'd like to add Amazon Linux to this list.  To do that, I had to reorg the checks so that `lsb_release` wasn't the first check.  
```
[ec2-user@ip-172-31-53-200 cfn-guard-lambda]$ diff Makefile Makefile.old
134,143c134
<         ifeq "$(shell cat /etc/system-release | grep -v 'Amazon Linux')" ""
< 			@echo Distro is Amazon Linux.
< 			sudo yum install jq gcc zip
<       else ifeq "$(shell cat /etc/system-release | grep -v 'CentOS Linux')" ""
< 			@echo Distro is CentOS.
< 			sudo yum install jq gcc zip musl-devel
<       else ifeq "$(shell cat /etc/system-release | grep -v 'Red Hat Enterprise Linux')" ""
< 			@echo Distro is RHEL.
< 			sudo yum install jq gcc zip musl-devel
<       else ifeq "$(shell lsb_release -si)" "Ubuntu"
---
>         ifeq "$(shell lsb_release -si)" "Ubuntu"
146a138,143
>       else ifeq "$(shell cat /etc/system-release | grep -v 'Red Hat Enterprise Linux')" ""
> 			@echo Distro is RHEL.
> 			sudo yum install jq gcc zip musl-devel
>       else ifeq "$(shell cat /etc/system-release | grep -v 'CentOS Linux')" ""
> 			@echo Distro is CentOS.
> 			sudo yum install jq gcc zip musl-devel
```
Can you change this to this updated block and make sure it still works for you?

```
pre-reqs:
	rustup target add x86_64-unknown-linux-musl
    ifeq ($(UNAME_S),Linux)
		@echo This is a $(UNAME_S) machine...
        ifeq "$(shell cat /etc/system-release | grep -v 'Amazon Linux')" ""
			@echo Distro is Amazon Linux.
			sudo yum install -y jq gcc zip
      else ifeq "$(shell cat /etc/system-release | grep -v 'CentOS Linux')" ""
			@echo Distro is CentOS.
			sudo yum install jq gcc zip musl-devel
      else ifeq "$(shell cat /etc/system-release | grep -v 'Red Hat Enterprise Linux')" ""
			@echo Distro is RHEL.
			sudo yum install jq gcc zip musl-devel
      else ifeq "$(shell lsb_release -si)" "Ubuntu"
			@echo Distro is Ubuntu
			sudo apt install jq musl-dev musl-tools zip awscli build-essential
			sudo ln -s /usr/bin/musl-gcc /usr/local/bin/x86_64-linux-musl-gcc
        endif
    endif
    ifeq ($(UNAME_S),Darwin)
		brew install filosottile/musl-cross/musl-cross
		ln -s /usr/local/bin/x86_64-linux-musl-gcc /usr/local/bin/musl-gcc
		brew install jq
    endif
```
I've had issues pasting this, so I'll just point out that there should be tabs in front of the shell commands and spaces in front of the make `ifeq`'s, etc:

```
^Irustup target add x86_64-unknown-linux-musl$
    ifeq ($(UNAME_S),Linux)$
^I^I@echo This is a $(UNAME_S) machine...$
        ifeq "$(shell cat /etc/system-release | grep -v 'Amazon Linux')" ""$
^I^I^I@echo Distro is Amazon Linux.$
^I^I^Isudo yum install -y jq gcc zip$
      else ifeq "$(shell cat /etc/system-release | grep -v 'CentOS Linux')" ""$
^I^I^I@echo Distro is CentOS.$
^I^I^Isudo yum install jq gcc zip musl-devel$
      else ifeq "$(shell cat /etc/system-release | grep -v 'Red Hat Enterprise Linux')" ""$
^I^I^I@echo Distro is RHEL.$
^I^I^Isudo yum install jq gcc zip musl-devel$
      else ifeq "$(shell lsb_release -si)" "Ubuntu"$
^I^I^I@echo Distro is Ubuntu$
^I^I^Isudo apt install jq musl-dev musl-tools zip awscli build-essential$
^I^I^Isudo ln -s /usr/bin/musl-gcc /usr/local/bin/x86_64-linux-musl-gcc$
        endif$
    endif$
    ifeq ($(UNAME_S),Darwin)$
^I^Ibrew install filosottile/musl-cross/musl-cross$
^I^Iln -s /usr/local/bin/x86_64-linux-musl-gcc /usr/local/bin/musl-gcc$
^I^Ibrew install jq$
    endif$
```

seawildau commented:
Addressed comments, (re)-tested pre-requisite installation and successfully built on `centos:latest`, `amazonlinux:latest` and `ubuntu:latest` docker images.
